modules/azure/src/main/java/org/testcontainers/azure/AzuriteContainer.java

```diff
@@ -0,0 +1,183 @@
+    private String host = DEFAULT_HOST;
+
+    private File cert = null;
+
+    private String certExtension = null;
+
+    private File key = null;
+
+    private String pwd = null;
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzuriteContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+    }
+
+    /**
+     * Configure SSL with a custom certificate and password.
+     *
+     * @param pfxCert  The PFX certificate file
+     * @param password The password securing the certificate
+     * @return this
+     */
+    public AzuriteContainer withSsl(final File pfxCert, final String password) {
+        cert = pfxCert;
+        pwd = password;
+        certExtension = ".pfx";
+        return this;
+    }
+
+    /**
+     * Configure SSL with a custom certificate and private key.
+     *
+     * @param pemCert The PEM certificate file
+     * @param pemKey  The PEM key file
+     * @return this
+     */
+    public AzuriteContainer withSsl(final File pemCert, final File pemKey) {
+        cert = pemCert;
+        key = pemKey;
+        certExtension = ".pem";
+        return this;
+    }
+
+    /**
+     * Sets the hostname we want to use to connect to our emulator. (default: {@link #DEFAULT_HOST})
+     *
+     * @param host The host name
+     * @return this
+     */
+    public AzuriteContainer withHost(final String host) {
+        this.host = host;
+        return this;
+    }
+
+    @Override
+    protected void configure() {
+        withEnv("AZURITE_ACCOUNTS", WELL_KNOWN_ACCOUNT_NAME + ":" + WELL_KNOWN_ACCOUNT_KEY);
+        withCommand(getCommandLine());
+        if (cert != null) {
+            final String certAbsolutePath = cert.getAbsolutePath();
+            logger().info("Using path for cert file: '{}'", certAbsolutePath);
+            withFileSystemBind(certAbsolutePath, "/cert" + certExtension, BindMode.READ_ONLY);
+            if (key != null) {
+                final String keyAbsolutePath = key.getAbsolutePath();
+                logger().info("Using path for key file: '{}'", keyAbsolutePath);
+                withFileSystemBind(keyAbsolutePath, "/key.pem", BindMode.READ_ONLY);
+            }
+        }
+        withExposedPorts(DEFAULT_BLOB_PORT, DEFAULT_QUEUE_PORT, DEFAULT_TABLE_PORT);
+    }
+
+    /**
+     * Returns the connection string for the default credentials.
+     *
+     * @return connection string
+     */
+    public String getDefaultConnectionString() {
+        return getConnectionString(WELL_KNOWN_ACCOUNT_NAME, WELL_KNOWN_ACCOUNT_KEY);
+    }
+
+    /**
+     * Returns the connection string for the account name and key specified.
+     *
+     * @param accountName The name of the account
+     * @param accountKey  The account key
+     * @return connection string
+     */
+    public String getConnectionString(final String accountName, final String accountKey) {
```

Comment:
let's return using the default account name and account key

Reply:
And what if the users decided to override the account configuration? 

I mean it is one thing that we are not helping them to do it but as you have pointed out in your other comment, if they don't like the defaults thy can still override things using the common Testcontainers features. If they do so, then they would not have a way to get the connection string from the container. So in my opinion it costs nothing to have this method and the users would be able to use it when they need to. What do you think?